### ISSUE TITLE
Multi deno.json resolver scopes fails for JSX

### ISSUE THREAD

**Issue by sant123** [denoland/deno#24293]
Version: Deno 1.44.4

From https://github.com/denoland/deno/pull/24206

I have a web application that renders HTML with hono/jsx. However I want to create some aliases for my JS client to use React. 

This is my project structure:

![image](https://github.com/denoland/deno/assets/7959437/29f245f1-3464-4c82-aba7-52c385040905)

deno.json (server)

```json
{
  "lock": false,
  "compilerOptions": {
    "jsx": "react-jsx",
    "jsxImportSource": "hono/jsx"
  },
  "imports": {
    "@unocss/core": "https://esm.sh/@unocss/core@0.58.0",
    "@unocss/preset-uno": "https://esm.sh/@unocss/preset-uno@0.58.0",
    "@unocss/reset": "https://esm.sh/@unocss/reset@0.58.0/tailwind-compat.css",
    "hono": "https://deno.land/x/hono@v3.11.7/mod.ts",
    "hono/html": "https://deno.land/x/hono@v3.11.7/helper/html/index.ts",
    "hono/jsx/jsx-runtime": "https://deno.land/x/hono@v3.11.7/jsx/jsx-runtime.ts",
    "$std/": "https://deno.land/std@0.209.0/"
  }
}

```

Notice how client.js is in red, this is because it can't resolve `react` which is expected:

![image](https://github.com/denoland/deno/assets/7959437/d2b74f84-b23f-4355-a169-73cb7c520427)

So I add a deno.json on the client:

```json
{
  "lock": false,
  "imports": {
    "react": "https://esm.sh/react@18.3.1"
  }
}

```

Now client.js is in green:

![image](https://github.com/denoland/deno/assets/7959437/01016c92-cc1f-483a-a6e7-19446fc3e282)

But Layout.tsx now fails because is loading React JSX by default but hono/jsx:

![image](https://github.com/denoland/deno/assets/7959437/21e0f439-4b07-49b0-8ec6-d2c9b08d191e)

Is there a way to apply JSX configuration only to the directory that holds deno.json on the client? Notice Layout.tsx is a folder upper static folder; so it should read the configuration of deno.json on the server.

![image](https://github.com/denoland/deno/assets/7959437/a83b66c2-2329-4592-8a6f-3542ea75c432)

Thanks.

**Comment by sant123** [comment]
FYI removing the deno.json on the client makes Layout.tsx work again:

![image](https://github.com/denoland/deno/assets/7959437/83498aa0-4da4-40bd-a432-01950f52d35b)

**Comment by nayeemrmn** [comment]
I can't reproduce this, the TSConfig including the jsx configuration should be sourced from the `deno.json` at the root directory. Can you show the contents of the output panel?

**Comment by nayeemrmn** [comment]
Okay I reproduced it, it's happening because the `static` subdirectory imports `react`, which pollutes the global type environment into using react instead of hono somehow. This may end up being blocked on https://github.com/denoland/vscode_deno/issues/488.

**Comment by sant123** [comment]
Thanks for the reply @nayeemrmn, so should this be closed as duplicate of https://github.com/denoland/vscode_deno/issues/488?

**Comment by nayeemrmn** [comment]
Let's keep tracking it here as well for searchability.

**Comment by KyleJune** [comment]
I'm having a similar issue even though I have a single deno.jsonc file. I verified it was related to the same commit by checking out the one before to verify I had no issue then, and checking out the commit for multi deno.json resolver scopes.

https://github.com/denoland/deno/issues/24342

**Comment by sant123** [comment]
With Deno v1.44.4 this is still failing for me. Please take a look:


https://github.com/denoland/deno/assets/7959437/e5fcc75f-bcfd-4fce-b486-9402492c44e7

**Comment by sant123** [comment]
Please @nayeemrmn could you take a look? Am I missing some configuration to make it work properly? 

Thanks 🙏🏼

**Comment by nayeemrmn** [comment]
@sant123 The feature hasn't released yet, try with `deno upgrade --canary`.

**Comment by sant123** [comment]
I knew I was missing something, it works like a charm with Deno canary:

https://github.com/denoland/deno/assets/7959437/906c784d-2f3e-49d4-8804-733774e0a6a5

Thank you so much @nayeemrmn !!!
